tests/polyskel_convex_test.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import division
+
+import pytest
+
+from ladybug_geometry.geometry2d.polygon import Polygon2D
+from ladybug_geometry.geometry2d.pointvector import Point2D
+from ladybug_geometry.geometry2d.line import LineSegment2D
+from ladybug_geometry import polyskel as lb_polyskel
+
+from pprint import pprint as pp
+import numpy as np
+
+# TODO Figure out if we need this later
+# # For comparison
+# import sys
+# paths = ['/app/polyskel/']
+# for path in paths:
+#     if path not in sys.path: sys.path.insert(0,path)
+# import polyskel as orig_polyskel
+
+#TODO: move to updated polyskel once we import lb-geom?
+def skeleton_edges(skeleton):
+    """
+    Consumes list of polyskeleton subtrees.
+    Skeleton edges are the segments defined by source point and each sink points.
+
+    Args:
+        skeleton: list of polyskel.Subtree, which are namedTuples of consisting of
+        a source point, and list of sink points.
+
+    Returns:
+        list of LineSegment2Ds
+    """
+    L = []
+    for subtree in skeleton:
+        source = subtree.source
+        for sink in subtree.sinks:
+            edge = LineSegment2D.from_end_points(
+                Point2D(source[0], source[1]),
+                Point2D(sink[0],sink[1])
+            )
+            L.append(edge)
+    return L
+
+# TODO Make this a from_array() method
+def helper_mtx2line2d(lst):
+    """ List of tuples of point coordinates to LineSegment2D"""
+    p1, p2 = lst[0], lst[1]
+    line2d = LineSegment2D.from_end_points(
+        Point2D(*p1), Point2D(*p2)
+        )
+    return line2d
+
+# TODO Make this a from_array() method
+def helper_mtx2polygon2d(lst):
+    polygon = Polygon2D([Point2D(*pt) for pt in lst])
+    return polygon
+
+
+def helper_assert_polygon_equality(polygon, chk_edges, holes=None, lb=True):
+    # Make into euclid classes
+    if holes is None:
+        holes = []
+    else:
+        holes = [helper_mtx2polygon2d(hole) for hole in holes]
+
+    polygon = helper_mtx2polygon2d(polygon)
+    chk_edges = [helper_mtx2line2d(edge) for edge in chk_edges]
+
+    #TODO: temp until we updted polyskel with lb geom?
+    vertices = [(v.x,v.y) for v in polygon.vertices]
+
+    # Run function
+    if lb:
+        skel = lb_polyskel.skeletonize(vertices, [])
+    else:
+        skel = orig_polyskel.skeletonize(vertices, [])
+    tst_edges = skeleton_edges(skel)
+
+    # Tests
+    # Check types
+    is_instant = isinstance(tst_edges, list)
+    for edge in tst_edges:
+        is_instant = is_instant and isinstance(edge, LineSegment2D)
+
+    if not is_instant:
+        print('Instants not match')
+        raise Exception('Instance error.')
+
+    # Check lines
+    # Assumes order is constant
+    is_equal = True
+    for chk_edge, tst_edge in zip(chk_edges, tst_edges):
+        is_equal = is_equal and np.allclose(
+            np.array(chk_edge.to_array()),
+            np.array(tst_edge.to_array()),
+            atol = 1e-10
+            )

```

Comment:
If this is all you are using numpy for, I'd much rather add an `is_equivalent` method to the `LineSegment2D` class that's a bit like the `Point2D.is_equivalent` method. Then, all you have to do here is:
```
is_equal = all(chk_edge.is_equivalent(tst_edge, 1e-10) for chk_edge, tst_edge in zip(chk_edges, tst_edges))
```

Reply:
Yeah I should be able to add that pretty easily. However, I think it would be useful too keep numpy as dev-requirement for future use cases. Essentially it's very powerful math library for nested lists (amongst other things), and I think a lot of it's functionality would improve tests in terms of speed and concision. I can flesh out my thoughts on this more in the campfire.

Reply:
To clarify, I'm definitely open to adding a `numpy` dependency if it consistently saves us from writing several lines of code. It's just that I don't think this is a case where it's really doing that.